.github/workflows/update-third-party-notices.yml

```diff
@@ -0,0 +1,33 @@
+name: Check and Update Third Party Notices
+on: workflow_call
+
+jobs:
+  license-check:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+          registry-url: "https://registry.npmjs.org"
+      - run: npm install -g generate-license-file@1
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          token: ${{ secrets.BOT_REPO_SCOPED_TOKEN }}
+      # Skip post-install scripts here, as a malicious
+      # script could steal the NPM_TOKEN.
+      - run: npm ci --ignore-scripts
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```

Comment:
since we're not installing any private npm packages, we don't need this npm token. if we remove it, we can just do an npm install without the `--ignore-scripts` flag and the rebuild step because we'd no longer have to worry about exposing the token